### PR TITLE
CLDR-13792 fix error messages for bad emoji

### PR DESCRIPTION
**Pull request by macchiati**
##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-13792
- [ ] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

File: tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCheckCLDR.java [hunk 1]
```diff
@@ -68,7 +71,7 @@
 
 public class TestCheckCLDR extends TestFmwk {
     static CLDRConfig testInfo = CLDRConfig.getInstance();
-    private final Set<String> eightPointLocales = new TreeSet<String>(
```
Comment (macchiati) [inline tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCheckCLDR.java:71]: The Eclipse reformatting cleans up some stuff like this.

File: tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCheckCLDR.java [hunk 2]
```diff
@@ -457,6 +460,52 @@ public void TestCheckNew() {
         errln("No failure message.");
     }
 
+    public void TestCheckNewRootFailure() {
```
Comment (macchiati) [inline tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCheckCLDR.java:463]: This is the only new part. We check to make sure we return errors on both E10-386 (root emoji codes) and ^^^

File: tools/java/org/unicode/cldr/test/CheckCLDR.java [hunk 3]
```diff
@@ -1094,10 +1094,11 @@ public final CheckCLDR check(String path, String fullPath, String value, Options
          */
         // if (value == cldrFileToCheck.getBaileyValue(path, null, null) && value != cldrFileToCheck.getWinningValue(path)) {
         if (value != null
-            && value.equals(cldrFileToCheck.getBaileyValue(path, null, null))
-            && !value.equals(cldrFileToCheck.getWinningValue(path))) {
-            return this;
+            && !value.equals(cldrFileToCheck.getWinningValue(path))
+            && cldrFileToCheck.getUnresolved().getStringValue(path) == null) {
```
Comment (macchiati) [inline tools/java/org/unicode/cldr/test/CheckCLDR.java:1098]: clean up the test. Better results (shouldn't have checked bailey value)

File: tools/java/org/unicode/cldr/test/CheckNew.java [hunk 4]
```diff
@@ -24,41 +24,27 @@ public CheckCLDR setCldrFileToCheck(CLDRFile cldrFileToCheck, Options options,
...
-        //        if (Phase.VETTING == getPhase()) {
-        //            setSkipTest(false); // ok
-        //        } else {
-        //            setSkipTest(true);
-        //            return this;
-        //        }
-
         super.setCldrFileToCheck(cldrFileToCheck, options, possibleErrors);
-        annotationsRoot = CLDRConfig.getInstance().getAnnotationsFactory().make("root", false);
+        isRoot = "root".equals(cldrFileToCheck.getLocaleID());
```
Comment (macchiati) [inline tools/java/org/unicode/cldr/test/CheckNew.java:28]: cleaned up this code. Simpler, since we just check the format of the value.

**Comment by macchiati** [comment]
See notes in file review.

File: tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCheckCLDR.java [hunk 5]
```diff
@@ -457,6 +460,52 @@ public void TestCheckNew() {
...
+        // Check that we get an error with the root value for an emoji.
+        final Factory annotationsFactory = testInfo.getCommonAndSeedAndMainAndAnnotationsFactory();
+        String locale = "quc";
```
Comment (pedberg-icu) [inline tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCheckCLDR.java:466]: Why are we using the locale code for Quiché here? A comment would help...

File: tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCheckCLDR.java [hunk 6]
```diff
@@ -457,6 +460,52 @@ public void TestCheckNew() {
...
+        CheckCLDR c = new CheckNew(annotationsFactory);
+        List<CheckStatus> result = new ArrayList<>();
+        Map<String, String> options = new HashMap<>();
+        for (Phase phase : Phase.values()) {
+            options.put(Options.Option.phase.getKey(), phase.toString());
+            for (String value : Arrays.asList("E10-836", CldrUtility.INHERITANCE_MARKER)) {
+
+                // make a fake locale
+                XMLSource rootSource = new SimpleXMLSource("root");
+                rootSource.putValueAtPath(path, "E10-836");
```
Comment (pedberg-icu) [inline tools/cldr-unittest/src/org/unicode/cldr/unittest/TestCheckCLDR.java:477]: Is "E10-836" the root annotation for @cp=\"😀\" ?? This is a bit cryptic...

**Comment by macchiati** [comment]
> Code looks good. Some comments might help explain why we are using locale "quc", value "E10-836", etc. Don't want to block merge on that however.

The latest PR fixes those issues (quc was a holdover, but the name doesn't matter), and also should make the tests pass.

**Comment by macchiati** [comment]
For some reason, Peter's approval isn't counting. Maybe github is getting
confused, since he commented earlier. Will try removing then re-adding him.

But in the meantime, Kristi, Tom, can one of you approve?

Mark

**Pull request merged**